code/modules/projectiles/projectile/bullets.dm

```diff
@@ -749,7 +749,7 @@
 /obj/item/projectile/bullet/invisible/on_hit(var/atom/target, var/blocked = 0) //silence the target for a few seconds on hit
 	if (..(target, blocked))
 		var/mob/living/L = target
-		if(L.silent && L.silent < 5)
+		if(!L.silent || L.silent && L.silent < 5)
```

Comment:
It should be logically equivalent to ```!L.silent || (L.silent && L.silent < 5)```

If you want to check silent is less than 5, you need it to not be null. Then you apply the silence.
The issue was silent is null by default, so L.silent would never return true and silence would never apply. This checks the null case.